docs/README.md

```diff
@@ -1,16 +1,48 @@
-# Developer-local docs build
+# Building documentation locally
+
+There are two methods to build the documentation, described below.
+
+In both cases, the generated output can be found in `generated/docs`.
+
+## Building in an existing Envoy development environment
+
+If you have an [existing Envoy development environment](https://github.com/envoyproxy/envoy/tree/master/bazel#quick-start-bazel-build-for-developers), you should have the necessary dependencies and requirements and be able to build the documentation directly.
 
 ```bash
 ./docs/build.sh
 ```
 
-The output can be found in `generated/docs`. By default configuration examples are going to be validated during build. 
-To disable validation, set `SPHINX_SKIP_CONFIG_VALIDATION` environment variable to `true`:
+By default configuration examples are going to be validated during build. To disable validation,
+set `SPHINX_SKIP_CONFIG_VALIDATION` environment variable to `true`:
 
 ```bash
 SPHINX_SKIP_CONFIG_VALIDATION=true docs/build.sh
 ```
 
+## Using the Docker build container to build the documentation
+
+If you *do not* have an existing development environment, you may wish to use the Docker build
+image that is used in continuous integration.
+
+This can be done as follows:
+
+```
+./ci/run_envoy_docker.sh 'ci/do_ci.sh docs'
```

Comment:
One extra ask. I'm about to add some additional validation stuff to the docs target for CI checking on external dependency metadata that feeds into docs. Can you make a fork of this `docs` target called `dev_docs` (or something like that), so that we don't have developers running the more expensive target locally?

Reply:
Its not actually necessary to use the `do_ci.sh` script at all - ie you can use `docs/build.sh` directly inside the container.

Ive updated to use `build.sh`